edge_utils.c

```diff
@@ -1554,6 +1593,57 @@ static void readFromTAPSocket(n2n_edge_t * eee) {
 
 /* ************************************** */
 
+#ifdef __ANDROID_NDK__
```

Comment:
Move this code to a separate file such as in tuntap for instance, and add empty hooks for platforms that do not need them. I would like to keep the code easy to read instead of having too many #ifdefs

Reply:
Are you suggesting to implement a callback api in n2n so an external code (e.g. the android code) can hook specific events (e.g. supernode registration updated?). Again, I find it difficult to turn some of this android specific code into api hooks, e.g. how would you name the hook at https://github.com/ntop/n2n/blob/81a3102661b8071b70686c9194740d2fbf40aaf1/edge_utils.c#L1116 ? Otherwise we can just define some android specific functions and call them in edge_utils.c under `#ifdef __ANDROID_NDK__` (e.g. `n2n_android_check_gateway` in the example above)